java/custom/com/sun/cldc/i18n/j2me/UTF_8_Writer.java

```diff
@@ -55,10 +54,61 @@
      * @exception  IOException  If an I/O error occurs
      */
     public void write(char cbuf[], int off, int len) throws IOException {
-        out.write(encodeUTF8(cbuf, off, len));
-    }
+        byte[] outputByte = new byte[4];     // Never more than 4 encoded bytes
+        int inputChar;
+        int outputSize;
+        int count = 0;
 
-    private native byte[] encodeUTF8(char cbuf[], int off, int len);
```

Comment:
There's the same problem as with UTF_8_Reader, we're calling the OutputStream::write method a bunch of times